### PR TITLE
Resources and scripts for sizing performance testing.

### PR DESCRIPTION
**Pull request by skordas**
WIP - PR open for discussion and suggestions.

### CODE REVIEW

File: tests/perfscale-sizing-recommendation/check_metrics.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#Check metrics used in the prometheus rules created for TempoStack. Refer issue https://issues.redhat.com/browse/TRACING-3399 for skipped metrics.
+# metrics="tempo_request_duration_seconds_count tempo_request_duration_seconds_sum tempo_request_duration_seconds_bucket tempo_build_info tempo_ingester_bytes_received_total tempo_ingester_flush_failed_retries_total tempo_ingester_failed_flushes_total tempo_ring_members"
+metrics="tempo_ingester_traces_created_total tempo_receiver_accepted_spans tempo_receiver_refused_spans"
+
+for metric in $metrics; do
+query="$metric"
+count=0
+    response=$(curl -s -k -H "Authorization: Bearer $TOKEN" -H "Content-type: application/json" "https://$THANOS_QUERIER_HOST/api/v1/query?query=$query")
+    # echo $response
```
Comment (iblancasa) [inline tests/perfscale-sizing-recommendation/check_metrics.sh:16]: Remove?

File: tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+metadata:
+  name: generate-traces
+  namespace: test-generate-traces
+spec:
+  completions: 50
+  parallelism: 50
+  template:
+    spec:
+      nodeSelector:
+        # node-role.kubernetes.io/workload: ""
```
Comment (iblancasa) [inline tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml:12]: Remove?

File: tests/perfscale-sizing-recommendation/00-before_test.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    sleep $sleep_time
+  fi
+done
+
+
+### PREPARE TEMPOSTACK ###
+log_task "TempoStack"
+oc create -f ./content/03-tempostack_small.yaml
+# oc create -f ./content/03-tempostack_medium.yaml
+# oc create -f ./content/03-tempostack_heavy.yaml
```
[the remark this answers is not in the document]
Comment (skordas) [reply tests/perfscale-sizing-recommendation/00-before_test.sh:97]: I was using this one script for all 3 cases (small, medium, heavy) depending which ENV was under the test I was uncommenting specific CR.
Suggestions:
- leave it at it is - add this step in README
- create 3 similar scripts (copy of this file) - 00_before_test_small, 00_before_medium, 00_before_heavy and create correct CR for that env.
- Change this based on parameter: ex. `sh 00_before_test.sh heavy`
- Even deeper - change this based on parameter, but not select file - but prepare CR (similar to `01-test.sh`)

File: tests/perfscale-sizing-recommendation/check_metrics.sh (hunk 1 above)
Comment (skordas) [reply tests/perfscale-sizing-recommendation/check_metrics.sh:16]: Can be removed. Used only for debugging.

File: tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml (hunk 2 above)
Comment (skordas) [reply tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml:12]: OK

File: tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+apiVersion: batch/v1
```
[the remark this answers is not in the document]
Comment (skordas) [reply tests/perfscale-sizing-recommendation/content/05-generate-traces.yaml:1]: It's based on test plan.
Duration - to have more than 1 hour of env under test.
Completions and parallelism - to run it from 50 nodes - at least simulate it.

**Comment by skordas** [comment]
@IshwarKanse PTAL

**Comment by IshwarKanse** [comment]
@skordas I'll get back to you on this PR

**Comment by IshwarKanse** [comment]
@skordas Thanks for putting this together. Sorry for the delay as I was caught up in the 3.3 release activities. The PR looks good to me, we can simplify it later by porting it to chainsaw tests. But for the time being this is good.

**Review by IshwarKanse — APPROVED** [review]

**Pull request merged**